presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -181,6 +182,19 @@ public HiveWriterFactory(
         this.pageSorter = requireNonNull(pageSorter, "pageSorter is null");
         this.maxSortFilesPerBucket = maxSortFilesPerBucket;
         this.immutablePartitions = immutablePartitions;
+        this.insertExistingPartitionsBehavior = HiveSessionProperties.getInsertExistingPartitionsBehavior(session);
+        if (immutablePartitions) {
+            checkArgument(
+                    insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ERROR ||
+                            insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.OVERWRITE,
+                    format("Unsupported insert existing partition behavior %s with immutable partitions", insertExistingPartitionsBehavior));
```

Comment:
I think it would be more readable to have something like this

`Presto is configured to treat Hive partitions as immutable. It is not allowed to set insert_existing_partitions_behavior to append.`